src/bin/jcli_app/utils/output_format.rs

```diff
@@ -0,0 +1,33 @@
+use serde::Serialize;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+pub struct OutputFormat {
+    /// Format of output data. Possible values: json, yaml.
+    /// Any other value is treated as a custom format using values from output data structure.
+    /// Syntax example: "{node_id}: up for {stats.uptime}".
+    /// Full syntax reference: https://docs.rs/tinytemplate/1.0.2/tinytemplate/syntax/index.html
+    #[structopt(long = "output-format", default_value = "yaml")]
+    format: String,
+}
+
+custom_error! { pub Error
+    YamlFormattingFailed { source: serde_yaml::Error } = "failed to format output as YAML",
+    JsonFormattingFailed { source: serde_json::Error } = "failed to format output as JSON",
+    CustomFormattingFailed { source: tinytemplate::error::Error } = "failed to format output as custom format",
+}
+
+impl OutputFormat {
+    pub fn format(&self, data: impl Serialize) -> Result<String, Error> {
+        Ok(match self.format.trim().to_ascii_lowercase().as_str() {
+            "yaml" => serde_yaml::to_string(&data)?,
+            "json" => serde_json::to_string_pretty(&data)?,
+            _ => {
+                let mut template = tinytemplate::TinyTemplate::new();
+                let template_name = "default";
+                template.add_template(template_name, &self.format)?;
+                template.render(template_name, &data)?
```

Comment:
Could this template engine be reused and not instantiated for every line of output?

Reply:
Must `TinyTemplate` always haul around a map of named templates, or is there a way to instantiate and use just one self-contained template?

Reply:
The engine is created and called at most once for the whole JSON fetched from REST. I think i see what you're thinking about: the template should be created during CLI args parsing, right?

Unfortunately yes, TinyTemplate always holds a whole collection of templates and can't be used without usage of keys.

Reply:
Ah, sorry, I thought it could be called repeatedly.

I wonder if there is some widespread template syntax that many people already know. Is this syntax mostly the same as Go templates?

Reply:
It's similar, but not more than probably 90% of templating tools, they're all basically the same